### DISCUSSION TITLE
Make things harder to delete in Coolify

### DISCUSSION THREAD

**Discussion by kazzkiq** [coollabsio/coolify#1814] [Feature Requests]
As of today is way too easy to delete things in Coolify.
This can be really dangerous for production projects. My proposal would be to:

## 1. Prevent accidents:

Create popups similar to when you try to delete a project in GitHub. Making clear the danger of the action.

<img src="https://github.com/coollabsio/coolify/assets/1953194/f360d73f-c802-4549-a690-be9ddf29464a" width="500">

## 2. Add security layer with 2FA:

Enable 2FA on user-level and if the user has it enabled, ask for 2FA code before deleting anything inside Coolify.

## 3. Communication of what would be lost:

Check if user is using shared mount disk and if the resources are going to be lost or will be available at disk somewhere. Communicate it in the popup from the 1st point.

e.g.:

> "You're deleting "my-cool-website" but it seems you have a shared volume at "/home/my-cool-website/images", some of your data may be available there even after deleting this resource."

**Comment by Wulfheart** [comment]
I think this would be a great addition to coolify.

**Comment by crisog** [comment]
Would love to see this implemented!

**Comment by peaklabs-dev** [comment]
@kazzkiq @Wulfheart @crisog This will be added via this PR https://github.com/coollabsio/coolify/pull/3024. The user has to confirm twice typing DELETE and with their password.

**Reply by crisog** [reply]
> @kazzkiq @Wulfheart @crisog This will be added via this PR #3024. The user has to confirm twice typing DELETE and with their password.

Great work!

**Comment by peaklabs-dev** [comment]
This will be available in the next version via password and text confirmation depending on the risk level. If you have further suggestions for improvements when the next version is released, please open a new discussion.
